### DISCUSSION TITLE
How to make my window on top after window.show()?

### DISCUSSION THREAD

**Discussion by Aues6uen11Z** [r0x0r/pywebview#1366] [Q&A]
Hello everyone, I’ve found that in the **Windows** system, when a window is displayed by calling `window.show()` from a hidden state, it does not become the focus. If there is another window in the focus state, the pywebview window will be obscured by that window. Here is an example:

```python
import time
import webview


def hide_show(window):
    for _ in range(3):
        time.sleep(3)
        window.hide()
        time.sleep(3)
        window.show()


if __name__ == '__main__':
    window = webview.create_window('Hide-show window', 'https://pywebview.flowrl.com/hello')
    webview.start(hide_show, window)
```

I’m not sure what the reasoning behind this is, but I would like my program to be on top and in focus every time it is displayed. Can someone help me? I would be very grateful.

**Comment by Aues6uen11Z** [comment]
I’ve discovered a very strange phenomenon. In fact, I wanted to minimize the program to the taskbar using pystray, which led to this issue. Currently, if I right-click the taskbar icon and click “Open,” then hide the window, cycling back and forth like this, the window consistently alternates between being on top and being obscured by other windows. However, if I directly left-click the taskbar icon (since default=True is set, a left-click is equivalent to a right-click followed by clicking Open—or so I thought), then subsequent opened windows will always be on top, except for the first time they are obscured. 

```python
"""
Run pywebview alongside with pystray to display a system tray icon.
"""

from threading import Thread

from PIL import Image
from pystray import Icon, Menu, MenuItem

import webview


if __name__ == '__main__':

    def on_open(icon, item):
        window.show()

    def on_exit(icon, item):
        icon.stop()
        window.destroy()    # can not completely exit
    
    def on_closing():
        t = Thread(target=window.hide)
        t.start()
        return False

    image = Image.open('logo/logo.png')
    menu = Menu(MenuItem('Open', on_open, default=True), MenuItem('Exit', on_exit))
    icon = Icon('Pystray', image, menu=menu)
    Thread(target=icon.run, daemon=True).start()

    window = webview.create_window('Webview', 'https://pywebview.flowrl.com/hello')
    window.events.closing += on_closing
    webview.start()
```

**Comment by XHLin-gamer** [comment]
Hi, Aues6uen11Z,

For the exit problem, i.e. the main thread do not exit after running the ```window.destroy()```, I think it is caused by the static return value within the ```on_closing()``` function. The main_thread, which created by ```webview.start()``` would not exit as long as the ```on_closing()``` function return ```False```. Hence, you could change the return value when you really need to exit the windows, like
```python 
"""
Run pywebview alongside with pystray to display a system tray icon.
"""

from threading import Thread

from PIL import Image
from pystray import Icon, Menu, MenuItem

import webview

# use signal to pass EXIT command
SIGNAL = {'on_running' : True}

if __name__ == '__main__':

    def on_open(icon, item):
        window.show()

    def on_exit(icon, item):
        icon.stop()
        # emit exit signal
        SIGNAL['on_running']= False
        window.destroy()    # can not completely exit
        
    def on_closing():
        t = Thread(target=window.hide)
        t.start()
        # use signal to exit window
        return not SIGNAL['on_running']

    
    image = Image.open('logo/logo.png')
    menu = Menu(MenuItem('Open', on_open, default=True),
                MenuItem('Exit', on_exit))
    icon = Icon('Pystray', image, menu=menu)

    Thread(target=icon.run, daemon=True).start()

    window = webview.create_window('Webview', 'https://pywebview.flowrl.com/hello')

    window.events.closing += on_closing

    webview.start()
```

**Reply by Aues6uen11Z** [reply]
Thanks a lot!

**Comment by brobin002** [comment]
Found a solution :)

try:
    if hasattr(window, 'handle') and window.handle:
        hwnd = window.handle
    else:
        hwnd = BrowserView.instances[window.uid].Handle.ToInt32()

    ctypes.windll.user32.SetForegroundWindow(hwnd)
    logging.info("window set to foreground")
except Exception as e:
    logging.error(f"window set to foreground : {e}")

**Comment by r0x0r** [comment]
I pushed a fix for this to master. Now window is activated when shown.

**Reply by brobin002** [reply]
I have tested and it's not fixing the issue. On first click the window still appears behind the other ones. My solution is working well on my side.
